src/run/run.c

```diff
@@ -1383,6 +1404,18 @@ static int transient_service_set_properties(sd_bus_message *m, const char *pty_p
                         return bus_log_create_error(r);
         }
 
+        if (arg_root_directory) {
+                _cleanup_close_ int fd = open_tree(AT_FDCWD, arg_root_directory, OPEN_TREE_CLONE|AT_RECURSIVE);
+                if (fd < 0)
+                        return log_error_errno(errno, "Failed to clone mount tree at '%s': %m", arg_root_directory);
+
+                r = sd_bus_message_append(m, "(sv)", "RootDirectoryFileDescriptor", "h", fd);
+                if (r < 0)
+                        return bus_log_create_error(r);
+
+                TAKE_FD(fd);
```

Comment:
this doesn't look right: last time i looked sd_bus_message_append() dups the fd anyway, hence the caller retains the original fd, and should close it. this line should go